### PR TITLE
[cxx-interop] Test reverse interop with std::span

### PR DESCRIPTION
**Pull request by susmonteiro**

### CODE REVIEW

**Comment by susmonteiro** [comment]
@swift-ci please smoke test

**Comment by susmonteiro** [comment]
@swift-ci please smoke test

File: test/Interop/CxxToSwiftToCxx/span/span-execution.cpp [hunk 1]
```diff
@@ -0,0 +1,133 @@
+// RUN: %empty-directory(%t)
+// RUN: split-file %s %t
+
+// RUN: %target-swift-frontend -typecheck %t/use-span.swift -typecheck -module-name UseSpan -emit-clang-header-path %t/UseSpan.h -I %t -enable-experimental-cxx-interop -Xcc -std=c++20 -clang-header-expose-decls=all-public -disable-availability-checking
```
Comment (egorzhdan) [inline test/Interop/CxxToSwiftToCxx/span/span-execution.cpp:4]: Do we rely on `-disable-availability-checking` here? Does the test fail with this flag removed? (here and on the next line)

Comment (susmonteiro) [reply test/Interop/CxxToSwiftToCxx/span/span-execution.cpp]: You're correct. It does not

**Comment by susmonteiro** [comment]
@swift-ci please smoke test

**Review by egorzhdan — APPROVED** [review]
LGTM, thanks!

**Comment by susmonteiro** [comment]
@swift-ci please smoke test

**Pull request merged**